src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/UserIdentityImpl.java

```diff
@@ -34,78 +34,88 @@ public class UserIdentityImpl implements UserIdentity {
     private final UserRolesCourthousesRepository userRolesCourthousesRepository;
 
     @SuppressWarnings({"PMD.AvoidDeeplyNestedIfStmts", "PMD.CyclomaticComplexity", "PMD.CognitiveComplexity"})
-    private String getEmailAddressFromToken() {
-        if (nonNull(SecurityContextHolder.getContext().getAuthentication())) {
-            Object principalObject = SecurityContextHolder.getContext()
-                .getAuthentication()
-                .getPrincipal();
-
-            if (principalObject instanceof Jwt jwt) {
-                Object emailsAddressesObject = jwt.getClaims().get(EMAILS);
-                if (emailsAddressesObject == null) {
-                    emailsAddressesObject = jwt.getClaims().get(PREFERRED_USERNAME);
+    private String getEmailAddressFromToken(Jwt token) {
+        Object principalObject = token;
+
+        if (principalObject instanceof Jwt jwt) {
+            Object emailsAddressesObject = jwt.getClaims().get(EMAILS);
+            if (emailsAddressesObject == null) {
```

Comment:
We have to add "email" claim as well which we have recently added. The sequence should be EMAILS -> EMAIL -> PREFERRED_USERNAME.